### DISCUSSION TITLE
Is there a way for the server to "notify" clients that there are changes to pull?

### DISCUSSION THREAD

**Discussion by MarkLFT** [Azure/azure-mobile-apps#651] [Q&A]
We are looking at using this tool for a Point of Sales type mobile app.  And a requirement is that one device is aware of invoices created by other devices in the same outlet.  So, I am wondering other than polling, is there some event or push message that could be used to inform clients that another device created an invoice, so we should do a pull to get the latest information?

**Comment by adrianhall** [comment]
This isn't a core feature of the SDK, but it's relatively easy to do.

1. Set up your transport - SignalR, WebSockets, or push notifications are all possible.
2. Trigger a broadcast message in your ASP.NET Core application / controller.  You can attach a standard ASP.NET Core filter to the controller to do this.
3. In your client, listen for the broadcast message and do a sync of the appropriate table (which should just be one entity).

There is some complexity to consider - you may not want to sync everything; you may not want all clients to sync at the same time; etc.  So it's not a slam-dunk, but you have all the capabilities to do this already.

**Reply by adrianhall** [reply]
new as of 6.0.5 - you can attach to the RepositoryUpdated event handler or use a PostCommitHookAsync() method inside an access control provider.

**Comment by FarisAhmed** [comment]
I miss this feature too and I hope it will be implemented in the future. Most frameworks today use WebSockets (like couchbase mobile) to provide real time data replication.
The important thing in case of web socket is the database ability to broadcast data changes to the sync service, without this database feature your sync service will not detect data changes in the BE database in real time. I implemented this in sql server and combined it with a websocket endpoint in the sync service.

**Reply by adrianhall** [reply]
I’d love to see the code - maybe you can write a blog post about it, or a longer “show and tell” post :-)

The main problem with putting that in the library is permissions.  How can I tell that a connected (via WebSockets) user should get any particular record that has been updated?  Permissions (and ensuring that users don’t get data they shouldn’t) is a big issue here for a generalized library.
